ablySpec/Auth.swift

```diff
@@ -529,5 +529,148 @@ class Auth : QuickSpec {
+                let tokenParams = ARTAuthTokenParams(clientId: expectedClientId)
+
+                rest.auth.createTokenRequest(tokenParams, options: options, callback: { tokenRequest, error in
+                    expect(tokenRequest?.clientId).to(equal(expectedClientId))
+                    expect(tokenRequest?.mac).toNot(beNil())
+                    expect(tokenRequest?.nonce).toNot(beNil())
+                })
+            }
+
+            // RSA9b
+            it("should support AuthOptions") {
+                let rest = ARTRest(options: AblyTests.commonAppSetup())
+                let auth: ARTAuth = rest.auth
+
+                let authOptions = ARTAuthOptions(key: "key:secret")
+
+                auth.createTokenRequest(nil, options: authOptions, callback: { tokenRequest, error in
+                    expect(tokenRequest?.keyName).to(equal("key"))
+                })
+            }
+
+            // RSA9c
+            it("should generate a unique 16+ character nonce if none is provided") {
+                let rest = ARTRest(options: AblyTests.commonAppSetup())
+
+                rest.auth.createTokenRequest(nil, options: nil, callback: { tokenRequest, error in
+                    expect(tokenRequest?.nonce.characters).to(haveCount(16))
+                })
+            }
+
+            // RSA9d
+            context("should generate a timestamp") {
+
+                it("from current time if not provided") {
+                    let rest = ARTRest(options: AblyTests.commonAppSetup())
+
+                    rest.auth.createTokenRequest(nil, options: nil, callback: { tokenRequest, error in
+                        expect(tokenRequest?.timestamp).to(beCloseTo(NSDate(), within: 1.0))
+                    })
+                }
+
+                it("will retrieve the server time if queryTime is true") {
+                    let rest = ARTRest(options: AblyTests.commonAppSetup())
+
+                    var serverTime: NSDate?
+                    waitUntil(timeout: testTimeout) { done in
+                        rest.time({ date, error in
+                            serverTime = date
+                            done()
+                        })
+                    }
+
+                    expect(serverTime).toNot(beNil(), description: "Server time is nil")
+
+                    let authOptions = ARTAuthOptions()
+                    authOptions.queryTime = true
+
+                    rest.auth.createTokenRequest(nil, options: authOptions, callback: { tokenRequest, error in
+                        expect(tokenRequest?.timestamp).to(beCloseTo(serverTime!, within: 6.0))
+                    })
+                }
+            }
+
+            // RSA9e
+            it("should TTL be optional and specified in milliseconds") {
+                let rest = ARTRest(options: AblyTests.commonAppSetup())
+
+                let tokenParams = ARTAuthTokenParams()
+                let expectedTtl = 6.0
+                tokenParams.ttl = expectedTtl
```

Comment:
@mattheworiordan Currently, `tokenParams.ttl` can not be nil. It is always initialised with the default value: 1 minute. Critical?